v3/src/components/graph/components/chartdots.tsx

```diff
@@ -121,59 +120,67 @@ export const ChartDots = memo(function ChartDots(props: IProps) {
-        } else {
-          return NaN
-        }
-      })
-      .attr(secondaryCenterKey, (anID: string) => {
-        if (cellIndices[anID] && secondaryScale) {
-          const {row} = cellIndices[anID],
-            {v} = cellIndices[anID].cell
-          return secondaryScale.range()[0] -
-            signForOffset * (v * primaryHeight + (row + 0.5) * pointDiameter + row * cellParams.overlap)
-        } else {
-          return NaN
+    const setPoints = () => {
+        const duration = enableAnimation.current ? transitionDuration : 0
+        selection
+          .transition()
+          .duration(duration)
+          .on('end', (id, i) => (i === selection.size() - 1) && onComplete?.())
+          .attr(primaryCenterKey, (anID: string) => {
+            if (cellIndices[anID]) {
+              const {column} = cellIndices[anID],
+                {h} = cellIndices[anID].cell
+              return baseCoord + signForOffset * ((h + 0.5) * primaryCellWidth) + (column + 0.5) * pointDiameter -
+                cellParams.numPointsInRow * pointDiameter / 2
+            } else {
+              return NaN
+            }
+          })
+          .attr(secondaryCenterKey, (anID: string) => {
+            if (cellIndices[anID] && secondaryScale) {
+              const {row} = cellIndices[anID],
+                {v} = cellIndices[anID].cell
+              return secondaryScale.range()[0] -
+                signForOffset * (v * primaryHeight + (row + 0.5) * pointDiameter + row * cellParams.overlap)
+            } else {
+              return NaN
+            }
+          })
+          .style('fill', (anID: string) => {
+            return (legendAttrID && dataConfiguration?.getLegendColorForCase(anID)) ?? defaultPointColor
+          })
+      },
+      onComplete = () => {
+        if (enableAnimation.current) {
+          enableAnimation.current = false
+          setPoints()
         }
-      })
-      .style('fill', (anID: string) => {
-        return (legendAttrID && dataConfiguration?.getLegendColorForCase(anID)) ?? defaultPointColor
-      })
+      }
+
+    setPoints()
   }, [dataConfiguration, primaryAttrPlace, secondaryAttrPlace, graphModel, dotsRef,
-            enableAnimation, primaryScale, primaryIsBottom, layout, secondaryAxisPlace,
-            computeMaxOverAllCells, primaryAttrID, secondaryAttrID, legendAttrID, dataset, secondaryScale])
+    enableAnimation, primaryScale, primaryIsBottom, layout, secondaryAxisPlace,
+    computeMaxOverAllCells, primaryAttrID, secondaryAttrID, legendAttrID, dataset, secondaryScale])
 
   useEffect(() => {
     select(dotsRef.current).on('click', (event) => {
       const element = select(event.target as SVGSVGElement)
       if (element.node()?.nodeName === 'circle') {
         const tItsID: string = element.property('id')
         const [, caseId] = tItsID.split("_")
-        dataset?.selectCases([caseId])
+        dataset?.setSelectedCases([caseId])
```

Comment:
This seems to suggest that every click will deselect all other points, regardless of whether the point is already selected (usually clicking on a selected point doesn't deselect other points) or whether any modifiers are down. Seems like some more logic would be required here.

Reply:
Yes, good catch. This slipped into this pull request by mistake. A subsequent pull request will handle this correctly.